### PR TITLE
chore(repo): auto approve Dependabot PR's

### LINKED ISSUE #112: Auto-Approval for Dependabot
Having to rubber stamp all the Dependabot issues to allow them to be merged is cumbersome. We need to integrate the [auto-approve GitHub action](https://github.com/marketplace/actions/auto-approve) in the aws-rfdk repo that will auto-approve any MR's created by Dependabot that successfully build.

### PR DESCRIPTION
**Pull request by horsmand**
- Adds support for using the auto approve GitHub action to automatically
approve PR's from Dependabot
- Adds support for using Dependabot to update the version of any GitHub
actions we are using
- Closes #112 

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: .github/workflows/auto-approve.yml
Comment (ryyakobe) [inline .github/workflows/auto-approve.yml:6]: Will this work only if all the **CI / build** steps for a PR were successful?

Comment (horsmand) [reply .github/workflows/auto-approve.yml:6]: This will give the "review approval" for dependabot PR's but those other checks will still need to run and be required to pass before merging becomes unblocked.

Comment (ryyakobe) [reply .github/workflows/auto-approve.yml:6]: Understood. Thanks.

**Review by ryyakobe — APPROVED** [review]
Looks good! Thanks, David!

**Comment by horsmand** [comment]
There is a limitation here where the GitHub action will only give 1 approval, but our repository requires 2 approvals for PR's to mainline. There's apparently no work-around: https://github.com/hmarr/auto-approve-action/issues/13

This still reduces the required approvals from 2 to 1 for dependabot PR's, which puts us in a better spot than we're in now. We'll just need to be cognizant of this and remember they'll still need a single approval from a team member.

**Review by jericht — APPROVED** [review]
LGTM

**Comment by ladhadeepak** [comment]
> There is a limitation here where the GitHub action will only give 1 approval, but our repository requires 2 approvals for PR's to mainline. There's apparently no work-around: [hmarr/auto-approve-action#13](https://github.com/hmarr/auto-approve-action/issues/13)
> 
> This still reduces the required approvals from 2 to 1 for dependabot PR's, which puts us in a better spot than we're in now. We'll just need to be cognizant of this and remember they'll still need a single approval from a team member.

Hello @horsmand,

Probably I'm missing something here. How have you worked around the problem of requiring just 1 approval instead of configured 2 reviewers? Also, would you require a team member to approve even with a dependabot ? Isn't it an auto-approve bot?

I have a similar problem - requires a minimum 2 approvals for a PR but wants a bot approval to be counted twice (or a configured value) so that a bot-approved PR can be auto-merged (without requiring any approval from a team member). Do you have any suggestions on it?

Thanks.

**Pull request merged**
